cloud/amazon/elastic_beanstalk.py

```diff
@@ -0,0 +1,457 @@
+#!/usr/bin/python
+#
+# This is a free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This Ansible library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = """
+---
+module: elastic_beanstalk
+short_description: Manage environments in Elastic Beanstalk.
+description:
+    - This module allows the user to create, terminate, or restart Elastic Beanstalk environments. >
+        This module has a dependency on python-boto.
+version_added: "2.0"
+author: "Alec Disharoon (adisharoon-socialware)"
+options:
+  application_name:
+    description:
+      - The name of the Elastic Beanstalk application
+    required: true
+    default: null
+  application_version:
+    description:
+      - The version of the Elastic Beanstalk application
+    required: false
+    default: null
+  environment_name:
+    description:
+      - The name of the Elastic Beanstalk environment
+    required: false
+    default: Randomly generated
+  cname:
+    description:
+      - The CNAME prefix to use for the Elastic Beanstalk environment
+    required: false
+    default: Randomly generated
+  application_s3_bucket:
+    description:
+      - The S3 bucket where the application file (WAR, docker container, etc.) is located
+    required: false
+    default: null
+  application_s3_key:
+    description:
+      - The S3 key for the application file (WAR, docker container, etc.)
+    required: false
+    default: null
+  solution_stack:
+    description:
+      - The solution stack to use when creating an environment
+    required: false
+    default: null
+  environment_options:
+    description:
+      - A list of environment option values as specified by the Elastic Beanstalk API-based CLI
+    required: true
+    default: null
+  state:
+    description:
+      - The desired state of the EB environment
+    required: true
```

Comment:
When I change this to `state=dict(default="present", choices=["present", "absent", "restarted"], required=True)`

It throws an error about required and default being mutually exclusive?
`fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "Module alias error: internal error: required and default are mutually exclusive for state"}`